### PR TITLE
Collection of refactors

### PR DESCRIPTION
**Pull request by Phil-Friderici**
Based on PR #1 this PR collects the refactors we want \o/

### CODE REVIEW

**Comment by Phil-Friderici** [comment]
After merging this, we should create a new major version :)

File: README.md [hunk 1]
```diff
@@ -11,11 +11,9 @@ as shown below.
 
 You can add any of the supported options for the types in this module. Please
 see the Puppet Labs
-[Type Reference](http://docs.puppetlabs.com/references/stable/type.html) for
+[Type Reference](https://www.puppet.com/docs/puppet/8/type.html) for
```
Comment (anders-larsson) [inline README.md:14]: Should we consider using 'latest' here to avoid forcing us to updating it once in a while?

**Review by anders-larsson — APPROVED** [review]
Looks OK to me.

I have a general question though. What's the point of wrapping all the resource type instead of calling the resource type directory? Is it the extra input validation? Wouldn't the source complain regardless if input is invalid?

File: README.md (hunk 1 above)
Comment (Phil-Friderici) [reply README.md:14]: Good point ! Updated the commit :)

**Comment by Phil-Friderici** [comment]
> I have a general question though. What's the point of wrapping all the resource type instead of calling the resource type directory? Is it the extra input validation? Wouldn't the source complain regardless if input is invalid?

I am not sure I did understand you correctly.
I guess the hashes in `init.pp` are usefull to be able to use hiera and it's merge functionality.

**Comment by anders-larsson** [comment]
> > I have a general question though. What's the point of wrapping all the resource type instead of calling the resource type directory? Is it the extra input validation? Wouldn't the source complain regardless if input is invalid?
> 
> I am not sure I did understand you correctly. I guess the hashes in `init.pp` are usefull to be able to use hiera and it's merge functionality.

I'm speaking about the wrapper defines for the resources. Like exec.pp, package.pp. E.g. Why do we have defines for reach resource instead of using create_resources($hash).

**Comment by Phil-Friderici** [comment]
Next guess, this was needed in the old times.
What you would like is available in one module from Voxpupuli:
https://github.com/voxpupuli/puppet-system
Instead of changing ours and migrating our hiera data, I would recommend to switch the modules.

**Comment by anders-larsson** [comment]
Looks like it has been archived? Do we know why?

**Comment by Phil-Friderici** [comment]
> Looks like it has been archived? Do we know why?

Missed that :( But I guess there are a lot more available giving this kind of functionality.
We could also add it to ours if you prefer.

**Comment by Phil-Friderici** [comment]
Released as v2.0.0-forked

**Comment by Phil-Friderici** [comment]
> Looks like it has been archived? Do we know why?

Maybe because [stdlib::manage](https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/REFERENCE.md#stdlib--manage) is good enough ?

**Comment by anders-larsson** [comment]
Hm. Maybe that's something that should be used instead. Indeed! Guess we'll still need this one for now but maybe we should move towards stdlib::manage.

**Pull request merged**